connector/saml/saml.go

```diff
@@ -263,6 +272,28 @@ func (p *provider) POSTData(s connector.Scopes, id string) (action, value string
 		return "", "", fmt.Errorf("marshal authn request: %v", err)
 	}
 
+	// Compress the request if redirect-binding is used
+	if p.redirectBinding {
+		var buf bytes.Buffer
+		zw, e := flate.NewWriter(&buf, flate.DefaultCompression)
+		if e == nil {
+			_, e = zw.Write(data)
+		}
+		if e == nil {
+			e = zw.Flush()
+		}
+		zw.Close()
+		if e != nil {
+			return "", "", fmt.Errorf("deflate request: %v", e)
+		}
+
+		redirectURL := fmt.Sprintf("%s?SAMLRequest=%s&RelayState=%s",
+			p.ssoURL,
+			url.QueryEscape(base64.StdEncoding.EncodeToString(buf.Bytes())),
+			url.QueryEscape(id))
+		return redirectURL, "", nil
```

Comment:
This method now returns completely different kind of values depending on how it's configured. The URL creation and HTML creation should be done in the same place.